File: devboxes/read-write-files/read_write_files.ts
import { Runloop } from "@runloop/api-client";

async function main() {
  // Initialize the Runloop client
  const runloop = new Runloop({
    bearerToken: process.env.RUNLOOP_API_KEY!,
  });

  // Create a devbox and wait for it to be ready
  const devbox = await runloop.devboxes.createAndAwaitRunning();
  console.log(`Created devbox with ID: ${devbox.id}`);

  // Write a small text file
  const filePath = "/home/user/example.txt";
  const content = "Hello from Runloop devbox!";
  await devbox.writeFileContents(filePath, content);
  console.log(`Wrote content to ${filePath}`);

  // Read the file back
  const readContent = await devbox.readFileContents(filePath);
  console.log(`Read content from ${filePath}: ${readContent}`);

  // Clean up
  await devbox.delete();
  console.log("Devbox deleted");
}

main().catch(console.error);

File: devboxes/read-write-files/read_write_files.py
import os
from runloop_api_client import Runloop

# Initialize the Runloop client
runloop_client = Runloop(bearer_token=os.environ.get("RUNLOOP_API_KEY"))

# Create a devbox and wait for it to be ready
devbox = runloop_client.devboxes.create_and_await_running()
print(f"Created devbox with ID: {devbox.id}")

# Write a small text file
file_path = "/home/user/example.txt"
content = "Hello from Runloop devbox!"
devbox.write_file_contents(file_path, content)
print(f"Wrote content to {file_path}")

# Read the file back
read_content = devbox.read_file_contents(file_path)
print(f"Read content from {file_path}: {read_content}")

# Clean up
devbox.delete()
print("Devbox deleted") 